core/src/main/java/com/linecorp/armeria/server/DefaultRoutingContext.java

```diff
@@ -133,6 +144,17 @@ public MediaType contentType() {
         return acceptTypes;
     }
 
+    @Override
+    public List<LanguageRange> acceptLanguages() {
+        return acceptLanguages.get();
+    }
+
+    @Nullable
+    @Override
+    public Locale getLocale(Collection<Locale> supportedLocales) {
```

Comment:
How about `selectLocale(Iterable<Locale>)` ?

Reply:
Agreed, I actually thought about this, but forgot about. I mostly put it there, because of the closeness to the media type parser.

It might even be preferable to have the `Accept` in `HttpRequest` as well, and just let `HttpRoutingContext` use that method, but that's a large change imho.

On the `selectLocale(Iterable)`, yes, much better name and signature.

Reply:
Alright, then let's move these methods to `HttpRequest`. `Accept` support would be even nicer, but it's up to you as always. It'd be nice if we could file an issue at least, so we can follow-up later.

Reply:
Sorry, but I don't have enough time to try to figure out routing internals right now. I'm putting it in `HttpHeadersBase` if that's ok, as that's where `contentType()` parsing is done, and adding a `default` method in HttpRequest (same as for `contentType()`). I tried putting the logic in `HttpRequest`, but unless I want to put a lot of logic into an interface, that will add a lot of duplicated logic as there are about 10 classes implementing that interface.

Btw, no on the `Iterable` unfortunately. `Locale.filter()` needs a `Collection`.

Reply:
> Btw, no on the Iterable unfortunately. Locale.filter() needs a Collection.

Then, how about copy the `supportedLocales` if it is not an instance `Collection`.
For example:

```java
Locale selectLocale(Iterable<Locale> supportedLocales) {
    Collection<Locale> locales;
	if (!supportedLocales instanceof Collection) {
		locales = supportedLocales;
    } else {
		locales = ImmutableList.copyOf(supportedLocales);
    }
}

Reply:
I somehow missed this comment. Going through all comments again.